pkgs/applications/graphics/inkscape/default.nix

```diff
@@ -48,32 +46,36 @@
 let
   python3Env = python3.withPackages
     (ps: with ps; [
+      # tomlq --slurp 'map(.tool.poetry.dependencies | to_entries | map(.key)) | flatten | map(ascii_downcase) | unique' $(fd pyproject.toml ~/Dokumente/repos/inkscape-extensions)
```

Comment:
Might be a good idea to add it to `passthru` so that people do not need to clone it manually. Also, this will ensure the correct version.

````diff
--- a/pkgs/applications/graphics/inkscape/default.nix
+++ b/pkgs/applications/graphics/inkscape/default.nix
@@ -44,6 +44,9 @@
 , wrapGAppsHook3
 , libepoxy
 , zlib
+, yq
+, fd
+, runCommand
 }:
 let
   python3Env = python3.withPackages
@@ -67,12 +70,12 @@ let
       pygobject3
     ] ++ inkex.propagatedBuildInputs);
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "inkscape";
   version = "1.3.2";
 
   src = fetchurl {
-    url = "https://inkscape.org/release/inkscape-${version}/source/archive/xz/dl/inkscape-${version}.tar.xz";
+    url = "https://inkscape.org/release/inkscape-${finalAttrs.version}/source/archive/xz/dl/inkscape-${finalAttrs.version}.tar.xz";
     sha256 = "sha256-29GETcRD/l4Q0+mohxROX7ciOFL/8ZHPte963qsOCGs=";
   };
 
@@ -182,7 +185,20 @@ stdenv.mkDerivation rec {
     done
   '';
 
-  passthru.tests.ps2pdf-plugin = callPackage ./test-ps2pdf-plugin.nix { };
+  passthru = {
+    tests.ps2pdf-plugin = callPackage ./test-ps2pdf-plugin.nix { };
+
+    pythonDependencies = runCommand "python-dependency-list" {
+      nativeBuildInputs = [
+        yq
+        fd
+      ];
+      inherit (finalAttrs) src;
+    } ''
+      unpackPhase
+        tomlq --slurp 'map(.tool.poetry.dependencies | to_entries | map(.key)) | flatten | map(ascii_downcase) | unique' $(fd pyproject.toml) > $out
+      '';
+  };
 
   meta = with lib; {
     description = "Vector graphics editor";
@@ -198,4 +214,4 @@ stdenv.mkDerivation rec {
       If you want to import .eps files install ps2edit.
     '';
   };
-}
+})
````

Reply:
Awesome idea, thank you!